### DISCUSSION TITLE
Banners customization

### DISCUSSION THREAD

**Discussion by Orsucciu** [Melledy/Grasscutter#2140] [General]
Hello,

Is there a breakdown of the banners syntax ? I'd like to set a custom banner that would allow to drop any existing character/weapon and i don't know what the values in the Banners.json mean, if someone already dug through this that would be helpful

Being able to load in a custom image and all would be great too

Thanks

**Comment by NotThorny** [comment]
Yes, it has been described in the [discord](https://discord.gg/T5vZU6UyeG) (I'd suggest you join there for actual help with most things)

Most of it is self-explanatory if you look at it, but what you want is to change is rateUpItems4/5 (event drops) or fallbackItems4/5Pool (non-rateup items).
Those don't have to be characters, and can instead be any item in the game. Weights are just [roll, percent without the decimal] (see discord for the full explanation of curve in resources channel)
You could instead use a simple banner editor like in [Jie's GCTools](https://github.com/jie65535/GrasscutterCommandGenerator/releases) for a much easier time. Set your current Banners.json then make whatever edits you want.

You can't load any custom assets with Banners.json, you want to use GIMI 3dmigoto to change that.
